controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -402,7 +401,11 @@ public void refresh() {
     }
 
     @Override
-    CompletableFuture<Data<Integer>> getStateData() {
+    CompletableFuture<Data<Integer>> getStateData(boolean ignoreCached) {
+        if (ignoreCached) {
+            cache.invalidateCache(statePath);
```

Comment:
One clarification about the semantics here, does `invalidateCache` reload the value? Otherwise, I suppose we will always be returning null in this method when `ignoreCached` evaluates to true.